python/pyarrow/parquet.py

```diff
@@ -101,14 +104,27 @@ def read(self, columns=None, nthreads=1):
         column_indices = self._get_column_indices(columns)
         if nthreads is not None:
             self.reader.set_num_threads(nthreads)
+
         return self.reader.read_all(column_indices=column_indices)
 
     def _get_column_indices(self, column_names):
         if column_names is None:
             return None
         else:
-            return [self.reader.column_name_idx(column)
-                    for column in column_names]
+            custom_metadata = self.metadata.metadata
+
+            # TODO(phillipc): Hack?
+            if custom_metadata and b'pandas' in custom_metadata:
```

Comment:
I think this is the last thing in this patch. I would like to have the option to ignore the metadata and read the file as-is as an Arrow table (without having the index columns tacked on against my will). So we can either add a `read_pandas` method to enables the metadata wrangling logic, or an option to `read` that does the same thing.